Automattic/newspack-network
# feat: add option to manually sync users

This PR adds an option to manually sync a user's data to the network. If the user doesn't exist in the network yet, it will also create them. Unlike the automatic syncing, this option will also sync a user's role. 

## Testing instructions

1. From your Hub site, edit one of your users; it's fine to start with one that's already been synced across your network. Confirm about halfway down the screen, under the "Newspack Network Tools" header, there should now be an option to manually sync a user: 

![image](https://github.com/Automattic/newspack-network/assets/177561/56145acf-0880-4831-b047-463b95201d36)

2. Change the user's role, and click Save. 
3. After saving, scroll back to the Newspack Network section, and click the 'Sync user across network' button. The page should refresh, and give you a success message at the top:

![image](https://github.com/Automattic/newspack-network/assets/177561/626f47c2-754d-4775-9db1-6ba356a65d4b)

4. Confirm that this creates a `network_manual_sync_user` action in your Hub's Event Log. 
5. On your Node site, confirm that the change has been made to the corresponding user, and that they have the new role. 
6. Create a new user on your Hub site, and edit them to add information (like a description, Facebook link, etc). Save.
7. After saving, on that new user's profile click the 'Sync user across network' button. 
8. Confirm that the new user is synced to your node sites with all of their populated information.
9. Repeat editing an existing user's role, saving, and manually syncing from one of your node sites. Confirm that the information is updated on the hub site.
10. Repeat creating a new user, saving, and manually syncing from one of your node sites. Confirm that the information is updated on the hub site.

Replaces #49 as the PR, just so I'm not putting words in Leo's mouth! 

See: 1206331646877116-as-1206522226618445
